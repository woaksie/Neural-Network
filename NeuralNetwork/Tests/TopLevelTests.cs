using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FakeItEasy;
using NeuralNetwork.Database;
using NeuralNetwork.DataPrep;
using NeuralNetwork.NetworkModels;
using NUnit.Framework;

namespace Tests
{
    [TestFixture]
    public class TopLevelTests
    {
        [Test]
        public void InitalCall()
        {
            PureData pureData = new PureData(new List<DataElement>());
            IInputsGeneratorFactory inputsFactory = A.Fake<IInputsGeneratorFactory>();
            ITargetsGeneratorFactory targetsFactory = A.Fake<ITargetsGeneratorFactory>();

            var sut = new DataPrepper(pureData, inputsFactory, targetsFactory);

            IList<DataSet> trainingData = sut.GetTrainingInputData();

            Assert.NotNull(trainingData);
            A.CallTo(() => inputsFactory.GetInputGenerators(pureData)).MustHaveHappened();
            A.CallTo(() => targetsFactory.GetTargetGenerators(pureData)).MustHaveHappened();
        }
    }
}
