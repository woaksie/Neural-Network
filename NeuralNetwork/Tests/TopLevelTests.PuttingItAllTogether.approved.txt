System.Collections.Generic.List`1[NeuralNetwork.NetworkModels.DataSet]